clients/fides-js/src/lib/initOverlay.ts

```diff
@@ -68,12 +68,38 @@ export const initOverlay = async ({
       // Determine which parent element to use as the container for rendering
       let parentElem;
       if (options.fidesEmbed) {
-        // Embed mode requires an existing element by which to embed the consent overlay
         parentElem = document.getElementById(FIDES_EMBED_CONTAINER_ID);
         if (!parentElem) {
-          throw new Error(
-            "Element with id fides-embed-container could not be found."
+          // wait until the hosting page's container element is available before proceeding in this script and attempting to render the embedded overlay. This is useful for dynamic (SPA) pages and pages that load the modal link element after the Fides script has loaded.
+          debugLog(
+            options.debug,
+            `Embed container not found (#${FIDES_EMBED_CONTAINER_ID}), waiting for it to be added to the DOM...`
           );
+          const checkEmbedContainer = async () =>
+            new Promise<void>((resolve) => {
+              const observer = new MutationObserver((mutations) => {
+                mutations.forEach((mutation) => {
+                  if (mutation.addedNodes.length) {
+                    parentElem = document.getElementById(
+                      FIDES_EMBED_CONTAINER_ID
+                    );
+                    if (parentElem) {
+                      debugLog(
+                        options.debug,
+                        `Embed container found (#${FIDES_EMBED_CONTAINER_ID}), continuing...`
+                      );
+                      observer.disconnect();
+                      resolve();
+                    }
+                  }
+                });
+              });
+              observer.observe(document.body, {
+                childList: true,
+                subtree: true,
+              });
+            });
+          await checkEmbedContainer();
```

Comment:
this only delays the **Overlay initialization**, _not_ the overall fides initialization. The window.Fides object is available while this waits.

Reply:
Are you sure? Don't we do `await initOverlay(...)` in the initialize call, which'll call `await renderFidesOverlay(...)` which then calls `await checkEmbedContainer()`?

Reply:
welp...turns out this is a race condition. If the document is in `loading` state by the time we initialize, `renderFidesOverlay` gets offset in the event and then `initOverlay` returns its promise (the `await` there apparently has no effect). If the document is `interactive` or `complete`, then the scenario you describe does happen because that `await` on `renderFidesOverlay` is in effect (in the `else` statement). So yeah, we just need to not await `renderFidesOverlay` which is working well in my testing.